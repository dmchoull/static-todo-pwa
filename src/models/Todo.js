import PropTypes from 'prop-types';

const Todo = (text, complete = false) => Object.freeze({ text, complete });

Todo.propTypes = PropTypes.shape({
  text: PropTypes.string.isRequired,
  complete: PropTypes.bool,
});

export default Todo;
