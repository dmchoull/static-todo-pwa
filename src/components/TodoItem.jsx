/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import { forbidExtraProps } from 'airbnb-prop-types';
import Todo from '../models/Todo';
import styles from '../assets/stylesheets/todo_list.pcss';

const TodoItem = ({ todo }) => (
  <li className={styles.todoItem}>
    <input className={styles.toggle} type="checkbox" />
    <label>{todo.text}</label>
  </li>
);

TodoItem.propTypes = forbidExtraProps({
  todo: Todo.propTypes.isRequired,
});

export default TodoItem;
