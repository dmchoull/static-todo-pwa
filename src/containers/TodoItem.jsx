import { connect } from 'react-redux';
import TodoItem from '../components/TodoItem';
import { toggleComplete } from '../actions/todo';

const mapDispatchToProps = dispatch => ({
  toggleComplete: todo => dispatch(toggleComplete(todo)),
});

export default connect(
  undefined,
  mapDispatchToProps,
)(TodoItem);
