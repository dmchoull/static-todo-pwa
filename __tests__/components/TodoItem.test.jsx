import React from 'react';
import { render } from 'react-testing-library';
import 'dom-testing-library/extend-expect';
import Todo from '../../src/models/Todo';
import TodoItem from '../../src/components/TodoItem';

describe('TodoItem', () => {
  test('renders a todo', () => {
    const { container } = render(<TodoItem todo={Todo('item 1')} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
