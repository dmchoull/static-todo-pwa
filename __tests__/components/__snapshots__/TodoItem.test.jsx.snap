// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders a todo 1`] = `
<li
  class="todoItem"
>
  <input
    class="toggle"
    type="checkbox"
  />
  <label>
    item 1
  </label>
</li>
`;
